train.py

```diff
@@ -235,7 +236,7 @@ def evaluate_in_parallel(best_model, cur_model, num_games, num_workers):
 
 
 
-def evolve(cur_model_path):
+def evolve(cur_model_path, other_opponent_for_selfplay):
```

Comment:
since we can take best model/other opponent/etc, maybe design a dictionary for passing in these, and document the keys somewhere